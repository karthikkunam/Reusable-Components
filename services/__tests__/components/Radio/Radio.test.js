import React from 'react';
import { shallow } from 'enzyme';
import Radio from '../../../src/components/Radio';

describe('Dropdown Component', () => {
  it('renders correctly', () => {
    const props = {
      radioOptions: [{}],
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<Radio {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders handleRadioClick correctly', () => {
    const props = {
      radioOptions: [
        {
          label: 'One',
          value: 1,
        },
        {
          label: 'Two',
          value: 2,
        },
        {
          label: 'Three',
          value: 3,
          disabled: true,
        },
      ],
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<Radio {...props} />);
    component
      .find('label')
      .at(0)
      .simulate('click', { handleRadioClick: () => {}, setValue: () => {} });
    expect(component).toMatchSnapshot();
  });
  it('renders Radio button with default value correctly', () => {
    const props = {
      radioOptions: [
        {
          label: 'One',
          value: 1,
        },
        {
          label: 'Two',
          value: 2,
        },
        {
          label: 'Three',
          value: 3,
          disabled: true,
        },
      ],
      defaultValue: 1,
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<Radio {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders Radio button with checked and disable', () => {
    const props = {
      radioOptions: [
        {
          label: 'One',
          value: 1,
        },
        {
          label: 'Two',
          value: 2,
        },
        {
          label: 'Three',
          value: 3,
          disabled: true,
          checked: true,
        },
      ],
      defaultValue: 1,
      onClick: jest.fn(),
      className: '',
      checked: false,
    };
    const component = shallow(<Radio {...props} />);
    expect(component).toMatchSnapshot();
  });
});
