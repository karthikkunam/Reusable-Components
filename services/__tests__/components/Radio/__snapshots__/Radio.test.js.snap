// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component renders Radio button with checked and disable 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      One
    </span>
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label "
    key="1"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Two
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label disabled"
    key="2"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Three
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Dropdown Component renders Radio button with default value correctly 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      One
    </span>
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label "
    key="1"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Two
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label disabled"
    key="2"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Three
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Dropdown Component renders correctly 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    />
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;

exports[`Dropdown Component renders handleRadioClick correctly 1`] = `
<fieldset
  className="form-group radio mt-1 "
>
  <label
    className="radio-label "
    key="0"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      One
    </span>
    <input
      checked="checked"
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label "
    key="1"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Two
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
  <label
    className="radio-label disabled"
    key="2"
    onClick={[Function]}
  >
    <span
      className="radio-text"
    >
      Three
    </span>
    <input
      checked=""
      name="radio"
      type="radio"
    />
    <span
      className="checkmark"
    />
  </label>
</fieldset>
`;
